test/TestDataFactory.js

```diff
@@ -0,0 +1,38 @@
+// ## TestDataFactory functions
+// Emulate old, pre-RDF/JS output (plain strings instead of Terms) to make tests pass
+
+// ### Creates an IRI
+function namedNode(iri) {
+  return iri;
+}
+
+// ### Creates a blank node
+function blankNode(name) {
+  return '_:b' + name;
+}
+
+// ### Creates a literal
+function literal(value, tag) {
+  if (tag) {
+    if (tag.length < 10) // hacky "heuristic" for test purposes only
+      return value + '@' + tag; // language tag
+    if (tag === "http://www.w3.org/2001/XMLSchema#integer")
+      return '"' + value + '"' + '^^' + tag; // treat integers specially as does SparqlGenerator
+    return value + '^^' + tag; // typed literal
+  }
```

Comment:
The above is not correct. It should be:
```
function literal(value, tag) {
  var suffix = '';
  if (tag)
    suffix = (/:/.test(tag) ? '^^' : '@') + tag
  return '"' + value + '"' + suffix;
}
```

That's how you go from value/tag to an old N3.js in-memory literal. The trick with the integers was actually masking a bug, which is that the last line of `unescapeString` should be `return string;`.

This fix then leads to a couple of bugs in `SparqlGenerator`, because that class has not yet been updated to RDF/JS.